### DISCUSSION TITLE
Can I set the order that string values are shown in a visualization?

### DISCUSSION THREAD

**Discussion by risafangs** [lightdash/lightdash#2560] [Q&A] [answered]
I have a simple Salesforce bar chart with counts of opportunities on the Y axis and the Salesforce stage on the X axis - things like `Qualifying`, `Needs Analysis`, `Negotiating`, `Closed Won`, etc. Would love to show them in pipeline stage order instead of descending by count. 

Is this possible to specify, or should do I need to do something like add a case statement to the SQL that numbers the stages? (eg `case when Qualifying then '0 Qualifying' when Needs Analysis then '1 Needs Analysis'` etc.

**Comment by TuringLovesDeathMetal** [comment, marked as answer]
Hi @risafangs - yes, you'd need to add something like a case statement to the SQL that numbers the stages for now. 

This is a really interesting question though! Typically, a funnel chart would only ever _**decrease**_ as you went through steps, but I'm assuming your chart doesn't? 🤔  

Out of interest, is there any other app you've used where you've been able to override the ordering in a funnel so it doesn't just order by decreasing size? Would love to see how other folks have done it!

**Reply by risafangs** [reply]
The Salesforce data usually only includes the "current" stage of the opportunity (vs multiple entries as one record goes through all the stages), so I can't rely on it to self-order. I'm thinking about graphing in pandas/seaborn where you can specify the order of values in your visualizations.

**Reply by TuringLovesDeathMetal** [reply]
Ah, makes sense. 

hmmm, that's a bit annoying that they don't give you multiple entries per record. But yes, either adding a `case` statement to append numbers to your steps and using Lightdash, or using pandas/seaborn where you can manually define the ordering might be the best options!
